earth_enterprise/src/common/serverdb/serverdbUtilities_unittest.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Open GEE Contributors
+// Unit tests for ServerdbUtilities.
+
+#include <gtest/gtest.h>
+#include <string>
+#include "common/serverdb/serverdbUtilities.h"
+
+
+class ServerdbUtilitiesTest : public testing::Test {
+protected:
+    virtual void SetUp() {}
+
+    virtual void TearDown() {
+    }
+
+    std::string GetJavaScriptString() {
+        std::string test_string =
```

Comment:
The use of a raw string might make this more readable.  See: https://en.cppreference.com/w/cpp/language/string_literal

Reply:
Good idea about the readability of the string. I wasn't crazy about what I had but now I've got it formatted and looking good.

I am looking at alternatives to using <regex>. It looks like for *nix <regex.h> can work. I am testing out a rough solution now. Having said that it won't work for windows.

Another option is bringing in boost regex or pcre.

Reply:
Yes I was going to sugest <regex.h> too.  But that is a good point about Windows.  We currently don't have boost as one of our third party libraries.  I would like to add it but that might increase this scope to add it now.   There are other libraries that would be nice to have access to in boost.

Reply:
What about using "#ifdef COMPILER_MSVC" and using <regex> with msvc, and then <regex.h> for every other situation?

This should cover all *nix and Windows.

Reply:
I think you have the a good idea there.  The one twist is Open GEE uses GCC for windows too *but* I believe it is a later version that has proper support for <regex>.  We could drive the decision to use <regex> vs <regex.h> based on compiler version.  CentOS 6 is the only supported OS that uses the older compiler when it builds.

Reply:
I think using regular expressions for parsing, or modifying a data structure is not a good approach in general. Regular expression don't offer a way to represent a complex data structure, such as a heterogenous map, or an array.  It is usually very easy to introduce a hard-to-spot bug in a regular expression, which only shows up on some occasions.

Reply:
@tst-dsugar I am in agreement that I think the better answer is refactoring and modifying the code to do what it says it does (read/write JSON).

Having said that, if the original idea is to keep backwards compatibility (returning JavaScript) then that constricts the fix a bit. If regex isn't a good approach, then either another third party library needs to be brought in that handles this situation, or a customized JSON parser has to be built for this specific problem set.